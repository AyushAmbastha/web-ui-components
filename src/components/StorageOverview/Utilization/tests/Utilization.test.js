import React from 'react';
import { shallow } from 'enzyme';

import { Utilization } from '../Utilization';
import { default as UtilizationFixtures } from '../fixtures/Utilization.fixture';

describe('<Utilization />', () => {
  it('renders correctly', () => {
    const component = shallow(<Utilization {...UtilizationFixtures.props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly in Loading state', () => {
    const component = shallow(<Utilization />);
    expect(component).toMatchSnapshot();
  });
});
