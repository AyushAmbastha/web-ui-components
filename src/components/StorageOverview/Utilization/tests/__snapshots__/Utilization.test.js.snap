// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Utilization /> renders correctly 1`] = `
<DashboardCard
  className={null}
>
  <DashboardCardHeader
    className={null}
  >
    <DashboardCardTitle
      className={null}
    >
      Utilization
    </DashboardCardTitle>
  </DashboardCardHeader>
  <DashboardCardBody
    LoadingComponent={[Function]}
    isLoading={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="kubevirt-utilization__time-duration kubevirt-utilization__time-duration--left"
        componentClass="div"
      >
        Time 
      </Col>
      <Col
        bsClass="col"
        className="kubevirt-utilization__time-duration kubevirt-utilization__time-duration--right"
        componentClass="div"
      >
        6 hour average
      </Col>
    </Row>
    <ChartAxis
      height={30}
      orientation="top"
      padding={
        Object {
          "bottom": 0,
          "left": 43,
          "right": 20,
          "top": 25,
        }
      }
      scale={
        Object {
          "x": "time",
        }
      }
      tickFormat={[Function]}
      tickValues={
        Array [
          1000000,
          1010000,
          1020000,
          1030000,
          1040000,
        ]
      }
    />
    <UtilizationBody>
      <UtilizationItem
        LoadingComponent={[Function]}
        data={
          Array [
            2,
            5,
            15,
            10,
            30,
          ]
        }
        decimalPoints={1}
        id="throughput"
        isLoading={false}
        maxY={30}
        title="Throughput"
        unit="Ki/s"
      />
      <UtilizationItem
        LoadingComponent={[Function]}
        data={
          Array [
            0,
            10,
            15,
            100,
            50,
          ]
        }
        decimalPoints={0}
        id="iops"
        isLoading={false}
        maxY={100}
        title="IOPS"
        unit="IOPS"
      />
      <UtilizationItem
        LoadingComponent={[Function]}
        data={
          Array [
            0,
            10,
            15,
            100,
            50,
          ]
        }
        decimalPoints={1}
        id="latency"
        isLoading={false}
        maxY={100}
        title="Latency"
        unit="ms"
      />
      <UtilizationItem
        LoadingComponent={[Function]}
        data={
          Array [
            2,
            5,
            15,
            10,
            30,
          ]
        }
        decimalPoints={1}
        id="recoveryRate"
        isLoading={false}
        maxY={30}
        title="Recovery rate"
        unit="Ki/s"
      />
    </UtilizationBody>
  </DashboardCardBody>
</DashboardCard>
`;

exports[`<Utilization /> renders correctly in Loading state 1`] = `
<DashboardCard
  className={null}
>
  <DashboardCardHeader
    className={null}
  >
    <DashboardCardTitle
      className={null}
    >
      Utilization
    </DashboardCardTitle>
  </DashboardCardHeader>
  <DashboardCardBody
    LoadingComponent={[Function]}
    isLoading={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="kubevirt-utilization__time-duration kubevirt-utilization__time-duration--left"
        componentClass="div"
      >
        Time 
      </Col>
      <Col
        bsClass="col"
        className="kubevirt-utilization__time-duration kubevirt-utilization__time-duration--right"
        componentClass="div"
      >
        6 hour average
      </Col>
    </Row>
    <ChartAxis
      height={30}
      orientation="top"
      padding={
        Object {
          "bottom": 0,
          "left": 43,
          "right": 20,
          "top": 25,
        }
      }
      scale={
        Object {
          "x": "time",
        }
      }
      tickFormat={[Function]}
      tickValues={null}
    />
    <UtilizationBody>
      <UtilizationItem
        LoadingComponent={[Function]}
        data={null}
        decimalPoints={1}
        id="throughput"
        isLoading={true}
        maxY={0}
        title="Throughput"
        unit="B/s"
      />
      <UtilizationItem
        LoadingComponent={[Function]}
        data={null}
        decimalPoints={0}
        id="iops"
        isLoading={true}
        maxY={0}
        title="IOPS"
        unit="IOPS"
      />
      <UtilizationItem
        LoadingComponent={[Function]}
        data={null}
        decimalPoints={1}
        id="latency"
        isLoading={true}
        maxY={0}
        title="Latency"
        unit="ms"
      />
      <UtilizationItem
        LoadingComponent={[Function]}
        data={null}
        decimalPoints={1}
        id="recoveryRate"
        isLoading={true}
        maxY={0}
        title="Recovery rate"
        unit="B/s"
      />
    </UtilizationBody>
  </DashboardCardBody>
</DashboardCard>
`;
